db/stats.py

```diff
@@ -1,120 +1,278 @@
+            _write_stats(connection, next_date, stats)
+            next_date = _get_next_hour(next_date)
+
+
+def _write_stats(connection, date, stats):
+    """Records a value with a given name and current timestamp."""
+    for name, value in six.iteritems(stats):
+        q = text("""
+            INSERT INTO statistics (collected, name, value)
+                 VALUES (:collected, :name, :value)""")
+        connection.execute(q, {"collected": date, "name": name, "value": value})
+
+
+def add_stats_to_cache():
+    """Compute the most recent statistics and add them to memcache"""
+    now = datetime.datetime.now(pytz.utc)
+    with db.engine.connect() as connection:
+        stats = _count_submissions_to_date(connection, now)
+        db.cache.set(STATS_MEMCACHE_KEY, stats,
+                     time=STATS_CACHE_TIMEOUT, namespace=STATS_MEMCACHE_NAMESPACE)
+        db.cache.set(STATS_MEMCACHE_LAST_UPDATE_KEY, now,
+                     time=STATS_CACHE_TIMEOUT, namespace=STATS_MEMCACHE_NAMESPACE)
+
+
+def get_stats_summary():
+    last_collected, stats = _get_stats_from_cache()
+    if not stats:
+        recent_database = load_statistics_data(1)
+        if recent_database:
+            stats = recent_database[0]["stats"]
+            last_collected = recent_database[0]["collected"]
+        else:
+            stats = {k: 0 for k in stats_key_map.keys()}
+    return stats, last_collected
 
-            result = connection.execute("SELECT lossless, count(*) FROM lowlevel GROUP BY lossless")
-            for row in result.fetchall():
-                if row[0]: stats_parameters['lowlevel-lossless'] = row[1]
-                if not row[0]: stats_parameters['lowlevel-lossy'] = row[1]
 
-            result = connection.execute("SELECT lossless, count(*) FROM (SELECT DISTINCT ON (mbid) mbid, lossless FROM lowlevel ORDER BY mbid, lossless DESC) q GROUP BY lossless;")
-            for row in result.fetchall():
-                if row[0]: stats_parameters['lowlevel-lossless-unique'] = row[1]
-                if not row[0]: stats_parameters['lowlevel-lossy-unique'] = row[1]
+def _get_stats_from_cache():
+    """Get submission statistics from memcache"""
+    stats = db.cache.get(STATS_MEMCACHE_KEY, namespace=STATS_MEMCACHE_NAMESPACE)
+    last_collected = db.cache.get(STATS_MEMCACHE_LAST_UPDATE_KEY,
+                                  namespace=STATS_MEMCACHE_NAMESPACE)
 
-            if update_db:
-                for key, value in six.iteritems(stats_parameters):
-                    connection.execute("INSERT INTO statistics (collected, name, value) VALUES (now(), %s, %s) RETURNING collected", (key, value))
+    return last_collected, stats
 
-            result = connection.execute("SELECT now()")
-            last_collected = result.fetchone()[0]
 
-        value = stats_parameters
+def format_statistics(data):
+    """
+    :param data: data from load_statistics_data
+    :return: statistics formatted for
```

Comment:
Also, for the sake of consistency we should decide on common way to document arguments, return values, and things like that. I tend to just use https://google.github.io/styleguide/pyguide.html#Comments, but I picked it just because it is easy to read. Maybe there's something better or more conventional.